client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/OpenSearchClientClassReplacement.java

```diff
@@ -71,13 +73,32 @@ private static Method retrieveMethod(Object openSearchClient, String id) {
     }
 
     private static void addOpenSearchInfo(String method, Object query, long executionTime) {
-        OpenSearchCommand info = new OpenSearchCommand(getIndex(query), method, query, executionTime);
+        OpenSearchCommand info = new OpenSearchCommand(getIndex(query), method, getQuery(query), executionTime);
         ExecutionTracer.addOpenSearchInfo(info);
     }
 
-    private static Object getIndex(Object query) {
+    private static Object getQuery(Object query) {
         try {
-            return query.getClass().getMethod(INDEX_METHOD).invoke(query);
+            return query.getClass().getMethod(GET_QUERY_METHOD).invoke(query);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            return null;
+        }
+    }
+
+    private static List<String> getIndex(Object query) {
+        try {
+            Object result = query.getClass().getMethod(INDEX_METHOD).invoke(query);
+            if (result == null) {
+                return null;
+            }
+
+            if (result instanceof List) {
+                return (List<String>) result;
+            } else if (result instanceof String) {
+                return Collections.singletonList((String) result);
+            } else {
+                return Collections.singletonList(result.toString());
```

Comment:
is this last case expected in normal behaviour? or is it just a safety check to not crash EM? in this latter case, might want to log such event (so we can see when/if it happens, and fix it if needed)

Reply:
I made it this way to consider all cases, but the normal behavior would be for the index to be a single String.

Reply:
Added logs